filament/backend/test/test_FeedbackLoops.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "BackendTest.h"
+
+#include <private/filament/EngineEnums.h>
+
+#include "ShaderGenerator.h"
+#include "TrianglePrimitive.h"
+
+#include <fstream>
+
+#ifndef IOS
+#include <imageio/ImageEncoder.h>
+#include <image/ColorTransform.h>
+
+using namespace image;
+#endif
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// Shaders
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+static std::string downsampleVs = R"(#version 450 core
+layout(location = 0) in vec4 mesh_position;
+void main() {
+    // Hack: move and scale triangle so that it covers entire viewport.
+    gl_Position = vec4((mesh_position.xy + 0.5) * 5.0, 0.0, 1.0);
+})";
+
+static std::string downsampleFs = R"(#version 450 core
```

Comment:
These shaders won't work on GLES/WebGL right? If so maybe make sure you don't build the test on those platforms? Or at least not run it on those platforms.

Reply:
The existing sister tests (eg `test_MRT.cpp`) use version 450 as well, and yeah they only run on macOS right now.